src/preferences/dialog/dlgpreflibrary.cpp

```diff
@@ -325,6 +326,11 @@ void DlgPrefLibrary::slotUpdate() {
                     kBpmColumnPrecisionConfigKey,
                     BaseTrackTableModel::kBpmColumnPrecisionDefault);
     spinbox_bpm_precision->setValue(bpmColumnPrecision);
+
+    // Rescan Library if a directory is relocated
+    if (m_bRelocatedDir) {
+        emit scanLibrary();
```

Comment:
I suggest to show a dialog like we do when a directory was added.
See slotHide, that needs to be structured a bit and the message text can be a variable which is selected based `m_bRelocatedDir` or `m_bAddedDirectory`.

Reply:
What is the need for a dialog?

Reply:
The same reason there's a dialog when adding a new directory. The user has to do it at some point, but we don't want to enforce the potentially very long scan NOW.

Reply:
Ah, I get what you're saying. Yeah, a dialog saying that is required that lets the user choose when to rescan would also work. It could have a button to rescan now or close the dialog without rescanning. The problem currently is that it isn't obvious why merely using the relink button doesn't make the tracks appear in the library again.

Reply:
Exactly, and we already have that dialog for added dirs, so we just need to adjust the message text for each case.

I'm curious what's happening under the hood during the migration  (I can't test the Flatpak migration myself atm).
I don't understand why the tracks don't show up.
They should be shown (but inaccessible) initially, and IIUC the relink adjusts the paths/loacations in the db, so they _should_ be accessible then.

Reply:
Relinking is needed when the pre-Flatpak database referred to files that are outside the Flatpak sandbox. Accessing them inside the sandbox requires going through the XDG File Chooser Portal. The path that Mixxx gets back from the portal is different from its real path on the underlying filesystem. Rather than `/home/user/dir1/dir2`, Mixxx gets back `/run/user/...`.

Reply:
Yes, I got that. So the rescan is required to get _per-track_ access via the sandbox?
I thought access right for the (each) parent directory is sufficient?

Reply:
Yes, permission for the parent directory seems to be sufficient. But when I tested, the tracks that appear to have been relocated need a rescan to show up in the library.